src/odemis/driver/technolution.py

```diff
@@ -129,155 +204,143 @@ def _setPixelSize(self, pixelSize):
         # NOTE: Do not write real username/password here since this is published on github in plaintext!
-        self.externalStorageURL_ftp = model.StringVA('ftp://username:password@example.com', setter=self.setFTP)
+        self.externalStorageURL = model.VigilantAttribute(urlparse('ftp://username:password@example.com/Pictures'),
+                                                          setter=self._setURL)
         self.acqDelay = model.FloatContinuous(2.0, range=(0, 100000), unit='s', setter=self._setAcqDelay)
 
         # Cell acquisition parameters
-        # TODO K.K. Perform type check on intput into the VA's floats and ints
-        self.cellTranslation = model.ListVA([[[50, 50]] * self._shape.value[0]] * self._shape.value[1])
-        # TODO K.K. Put a check according to this input in the assemble_mega_field_check whih includes the counting up
-        #  per list. Then delete the dupplications below. (counting up allows to visually inspect te coordinate shifts)
-        self.cellTranslation = model.ListVA([[[10 + j, 20 + j] for j in range(i, i + self._shape.value[0])]
-                                             for i in range(0, self._shape.value[1] * self._shape.value[0],
-                                                            self._shape.value[0])])
-        self.celldarkOffset = model.ListVA([[0] * self._shape.value[0]] * self._shape.value[1])
-        self.celldigitalGain = model.ListVA([[1.2] * self._shape.value[0]] * self._shape.value[1])
-        self.celldigitalGain = model.ListVA([[j for j in range(i, i + self._shape.value[0])]
-                                             for i in range(0, self._shape.value[1] * self._shape.value[0],
-                                                            self._shape.value[0])])
+        self.cellTranslation = model.ListVA([[[50, 50]] * self._shape[0]] * self._shape[1],
+                                            setter=self._setCellTranslation)
+        self.cellDarkOffset = model.ListVA([[0] * self._shape[0]] * self._shape[1], setter=self._setcellDarkOffset)
+        self.cellDigitalGain = model.ListVA([[1.2] * self._shape[0]] * self._shape[1], setter=self._setcellDigitalGain)
         self.cellCompleteResolution = model.ResolutionVA((800, 800), ((10, 10), (1000, 1000)))
 
+        # TODO K.K. pass right metadata from new simulator
+
+        # Setup hw and sw version
+        # TODO make call set_system_sw_name to new simulator (if implemented)
+        self._swVersion = self._swVersion + ", " + "PUT NEW SIMULATOR DATA HERE"
+        self._hwVersion = self._hwVersion + ", " + "PUT NEW SIMULATOR DATA HERE"
+
         # Gather metadata from all related HW components and own _meta_data
         self.md_devices = [self._metadata, self.parent._mirror_descanner._metadata,
                            self.parent._ebeam_scanner._metadata]
-        self._metadata[model.MD_HW_NAME] = "MPPC"
+        self._metadata[model.MD_HW_NAME] = "MPPC" + "/" + name
```

Comment:
In case the name of the comonent is defined differently during initialization this is added. So if the brand name of the mppc is foobar the result would be MPPC/foobar. Or does the entered name always be MPPC?

Reply:
I think one or the other is enough. The API indeed does not allow us to check the HW for the mppc. Only for the SAM and ASM. We could check with Robert if we have a more specific model instead of just mppc. I would not add name as this is the descriptive one from the yaml file.

Reply:
I agree with Sabrina that adding to HW_NAME the name from the YAML file is not useful.